### ISSUE TITLE
Samples Story Arc Proposal

### ISSUE THREAD

**Issue by ryanvolum** [dapr/dapr#218]
## Samples Story Arc Proposal
This discussion is a product of feedback from very early adopters (OCTO devs) about the onboarding experience:

1. **Problem**: As it stands, developers aren't sure where to start (standalone vs k8s). If they start with kubernetes, they need substantial background knowledge. Further, setting up a cluster/state store takes time, preventing a "5 minutes to wow" experience. 

     

- [x] **Proposal**: Guide users to start with the standalone approach, then introduce samples by level of difficulty/complexity:

     - Number the samples, rename current `local_zero_to_hero` to `1. Hello World` and the current `Kubernetes_zero_to_hero` to `2. Hello Kubernetes`
     - The Hello World sample should start by introducing Actions running locally, and indicate that if developers want to use Kubernetes they can see sample 2
     - The Hello World sample should end with a **Next Steps** section that includes a link to sample 2, "deploy this application on Kubernetes"
     - The Hello Kubernetes sample should start with a note about how it uses the same services from sample 1, but doesn't require the user to have gone through sample 1.
     
2. **Problem**: The distributed calculator sample only runs on Kubernetes. How would I run this locally? 

     

- [ ] **Proposal:** Each sample should run seamlessly locally or on k8s. We should not double the number of samples by creating a local/k8s version of each.

     - The Hello World samples should be using the same code/microservices. Note that this is currently blocked due to https://github.com/actionscore/cli/issues/35. For now, the local sample can only have a single microservice.
     - Each additional sample (e.g. distributed calculator, pub/sub, etc.) should include the artifacts necessary to deploy it locally or in K8s. Their READMEs should have a local deployment section _and_ a k8s deployment section. This might be somewhat awkward, since each approach does have some of its own artifacts (e.g. Dockerfile, .yaml files), but we'll mitigate this by steering into it in the READMEs (e.g. "Note that you do not need the .yaml files or Dockerfile to run this sample locally").

3. **Problem**: The current samples take too long to get running.

     

- [ ] **Proposal**: Aim for <10 minutes to get the sample running, then explain the code/artifacts

     - Start by showing the steps to run, then explain what they do/how the code works (right now explanations are intermixed with instructions)
     - Consider including scripts to get the K8s samples running quickly (instead of hand-deploying each microservice)

4. **Problem**: The current samples don't offer a consistent structure or conventions for Actions projects

     

- [ ] **Proposal**: Create a spec for the anatomy of an Actions project

     - Decide on a naming convention (for at least the samples to use)
     - Determine a directory structure (for at least the samples to use)

**Comment by msfussell** [comment]
@pruthvidhodda - Can you look at this and take what is necessary for the samples plan and then close this. There are some things in here that are needed to carry over to the samples plan.

**Comment by pruthvidhodda** [comment]
@msfussell I have added the following points to the samples plan:

1. Every sample should run locally and in k8s mode and instructions for both should be in the same 	sample (except for the hello-world sample).  

2. Every sample should first have instructions to run the sample and then have a section explaining 	how it works. 

Added corresponding work items to make samples consistent with these points.
